fdbclient/DatabaseConfiguration.cpp

```diff
@@ -500,23 +501,23 @@ void DatabaseConfiguration::overwriteProxiesCount() {
 		    .detail("MutableGrvCPCount", mutableGrvProxyCount)
 		    .detail("MutableProxiesCount", mutableProxiesCount);
 
-		if (grvProxyCount == -1 && commitProxyCount > 0) {
-			if (mutableProxiesCount > commitProxyCount) {
-				grvProxyCount = mutableProxiesCount - commitProxyCount;
+		if (mutableGrvProxyCount == -1 && mutableCommitProxyCount > 0) {
```

Comment:
I think it make sense to replace the `DatabaseConfiguration::XXXProxyCount` usage with `mutable` values to avoid order of applying keys to DB impacting the outcome from this routine, hence, making the function _deterministic_

One request: can you please add sufficient code documentation to highlight this fact, in someways it will also help anyone updating the code in future to be aware of all necessary nuances. 

Thanks.

Reply:
https://github.com/apple/foundationdb/issues/6668 to address the problem of this code being tricky to understand.